hal/network/lwip/cellular/pppncpnetif.cpp

```diff
@@ -254,6 +254,21 @@ void PppNcpNetif::ncpEventHandlerCb(const NcpEvent& ev, void* ctx) {
         const auto& cev = static_cast<const CellularNcpAuthEvent&>(ev);
         LOG(TRACE, "New auth info");
         self->client_.setAuth(cev.user, cev.password);
+    } else if (ev.type == NcpEvent::NCP_STATE_CHANGED) {
+        const auto& cev = static_cast<const NcpStateChangedEvent&>(ev);
+        if_event evt = {};
+        struct if_event_power_state ev_if_power_state = {};
+        evt.ev_len = sizeof(if_event);
+        evt.ev_type = IF_EVENT_POWER_STATE;
+        evt.ev_power_state = &ev_if_power_state;
+        if (cev.state == NcpState::ON) {
+            evt.ev_power_state->state = IF_POWER_STATE_UP;
+            LOG(TRACE, "NCP power state changed: IF_POWER_STATE_UP");
+        } else {
```

Comment:
If we take a look at `off()` implementation, we can see that it's still possible that the NCP client will go into an off state even if the modem fails to power down.

Reply:
Can `modemHardReset(true)` turn off the modem reliably? I do see it is invoked if modem fails to be powered off:
```
        int rv = modemPowerOff();
        if (rv != 0) {
            modemHardReset(true);
        }
        ncpState(NcpState::OFF);
```

Reply:
Supposedly, however ideally we should still check the current modem state (`modemPowerState()`).

Reply:
On boot with `SYSTEM_MODE(AUTOMATIC)` or `MANUAL` the NCP client will also default to `OFF` state without reflecting the actual power state of the modem.

Reply:
But in `system_power_manager`, the `on` of `InterfaceRuntimeState` reflect the initial state of modem power and only when `SaraNcpClient::ncpState(NcpState state)` is called the `on` variable will be updated, despite the current value of `ncpState_` in `sara_ncp_client.cpp`.

Reply:
Sure, and by default it would be assumed that the modem is off, where in fact it might be on.

Reply:
Then we should initialize `ncpState_` accordingly in certain init function by checking the modem state, either through AT command or reading `UBVINT`.